R/syncBP.R

```diff
@@ -0,0 +1,164 @@
+#' 
+#' The NF-OSI workflow produces a single merged maf file that represents a filtered subset of the `maf`s,
+#' containing only the (non-germline) data that _can_ be released for cBioPortal.
+#' However, this data file by itself is not immediately loadable into an instance of a cBioPortal server
+#' and needs to be packaged with other files, such as this 
+#' [example of a public mutations dataset](https://github.com/cBioPortal/datahub/tree/1e03ea6ab5e0ddd497ecf349cbee7d50aeebcd5e/public/msk_ch_2020). 
+#' This is a wrapper that goes through several steps needed to create said bundle of cBioPortal files more conveniently.
+#' 
+#' 1. A simple sanity check that this is the version of the maf release file that we want, based on the samplesheet.
+#' For example, version 1 of the samplesheet will generate a version 1 of the merged maf, 
+#' but if there is a later correction that retracts a sample (`is_releasable`=FALSE), 
+#' that step of the workflow to generate merged maf will be rerun (or it should be), so we'll want to make sure the latest versions of these files are used.  
+#' The latest versions of the samplesheets tied to each release are currently stored in `syn38793855`. 
+#' (Note: Please file issue to update this doc if this changes.)
+#' 
+#' 2. Make the clinical data files. 
+#' For NF, the clinical metadata are annotations on the files/surfaced in a view and are pretty basic. 
+#' In the future, it would be preferable to store clinical metadata in a real normalized table.
+#' For now, this clinical data is pulled in from the view. To map NF clinical variables to the 
+#' [cBioPortal dictionary](https://github.com/cBioPortal/clinical-data-dictionary/blob/e9ec08f48bd57aabf193da70cdb5b88bdef5d01d/docs/resource_uri_to_clinical_attribute_mapping.txt)
+#' [as recommended](https://docs.cbioportal.org/file-formats/#custom-columns-in-clinical-data), 
+#' this step requires a `ref_map`, which is a YAML file.
+#' 
+#' 3. Make meta files. Meta files are needed for describing the study, mutations data file, clinical data files. 
+#' 
+#' @param merged_maf Synapse id of `merged maf` file for public release.
+#' @param samplesheet Synapse id or local path to samplesheet with release info.
+#' @param ref_map YAML file specifying the mapping of (NF) clinical metadata to cBioPortal model. See details.
+#' @param ref_view A view that contains clinical data for the release files. 
+#' @param cancer_study_identifier The study identifier, defaults to `nfosi_YEAR`.
+#' @param publish_dir Where to output the set of files. 
+#' Defaults to (creating if necessary) a folder with same name as `cancer_study_identifier`.
+#' @param verbose Whether to provide informative messages throughout.
+#' @export
+syncBP_maf <- function(merged_maf,
+                       samplesheet,
+                       ref_map,
+                       ref_view, 
+                       cancer_study_identifier = glue::glue("nfosi_{format(Sys.Date(), '%Y')}"),
```

Comment:
Is this something that is displayed on cBioPortal? One challenge we have faced in the past is ensuring researchers get credit for their data. 


```suggestion
                       cancer_study_identifier = glue::glue("nfosi_{format(Sys.Date(), '%Y')}"),c
```

Reply:
No, it's the `name` and  `short_name` fields that are displayed. However, that reminds me to update the file with a better `citation` option. I'll confirm with @jaybee84 what name and reference we're actually using for display of this dataset.

Reply:
Sounds good!